# Supermarket Project

Provide the summary statistics of the following variables:

<li>Income (in £) </li> <li> Express store - Total spending</li>
<li>Express store - No. of transactions</li>
<li>Superstore – No. of transactions</li>
<li>Superstore – Total Spending</li>
<li>Females (create a new dummy variable from the ‘Gender’ variable)</li>
<li>London (create a new dummy variable from the ‘Region’ variable)</li>

In [1]:
#Summary of Statistics

# - Measures of location: Mean,median, mode
# - Measures of spread: Range, interquartile range, standard deviation (SD)

In [2]:
import numpy as np
import pandas as pd

In [3]:
df = pd.read_excel("Part_A STATSTICS AND DATA MODELLING COURSEWORK.xlsx")
df

customer.id  express.no.transactions  express.total.spend  \
0         1668.0                       54              1559.20   
1         1827.0                       59               619.05   
2         3003.0                       59               211.50   
3         4398.0                       58              1118.04   
4         5055.0                       47              1318.61   
..           ...                      ...                  ...   
96       94420.0                       57              1144.88   
97       95503.0                       46                63.14   
98       97005.0                       23                69.48   
99       98018.0                       52              2494.40   
100          NaN                     5074             83495.71   

     superstore.no.transactions  superstore.total.spend  gender        income  \
0                            60                30874.62  Female  1.890172e+04   
1                            78                36466.45    Male  2.446498e+04   
2                            91                45903.40    Male  3.126321e+04   
3                            70                32175.17    Male  1.974213e+04   
4                            62                31109.30    Male  1.616582e+04   
..                          ...                     ...     ...           ...   
96                           72                32005.10  Female  2.379603e+04   
97                           89                52975.30    Male  3.834971e+04   
98                           99                52350.80  Female  3.377846e+04   
99                           49                30209.20  Female  1.255140e+04   
100                        7559              3892480.32     NaN  2.503501e+06   

    affluency              Region  
0         Mid       West Midlands  
1        High          North West  
2        High          North West  
3         Low  Yorkshire & Humber  
4         Mid              London  
..        ...                 ...  
96        Mid              London  
97        Low              London  
98        Mid       East Midlands  
99        Low              London  
100         0                   0  

[101 rows x 9 columns]

A quick scan of the data above shows the 100th row to have ouliers and object columns empty. This row appears to be the sum total of the data set so we would be skipping it by passing in **skipfooter**  while importing our excel file again.

In [4]:
df = pd.read_excel("Part_A STATSTICS AND DATA MODELLING COURSEWORK.xlsx",skipfooter=1)
df

customer.id  express.no.transactions  express.total.spend  \
0          1668                       54              1559.20   
1          1827                       59               619.05   
2          3003                       59               211.50   
3          4398                       58              1118.04   
4          5055                       47              1318.61   
..          ...                      ...                  ...   
95        93783                       39                48.04   
96        94420                       57              1144.88   
97        95503                       46                63.14   
98        97005                       23                69.48   
99        98018                       52              2494.40   

    superstore.no.transactions  superstore.total.spend  gender        income  \
0                           60                30874.62  Female  18901.724734   
1                           78                36466.45    Male  24464.981612   
2                           91                45903.40    Male  31263.210081   
3                           70                32175.17    Male  19742.126261   
4                           62                31109.30    Male  16165.819721   
..                         ...                     ...     ...           ...   
95                          92                53710.83  Female  35899.146160   
96                          72                32005.10  Female  23796.034048   
97                          89                52975.30    Male  38349.705738   
98                          99                52350.80  Female  33778.457469   
99                          49                30209.20  Female  12551.396826   

   affluency              Region  
0        Mid       West Midlands  
1       High          North West  
2       High          North West  
3        Low  Yorkshire & Humber  
4        Mid              London  
..       ...                 ...  
95       Low          North West  
96       Mid              London  
97       Low              London  
98       Mid       East Midlands  
99       Low              London  

[100 rows x 9 columns]

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   customer.id                 100 non-null    int64  
 1   express.no.transactions     100 non-null    int64  
 2   express.total.spend         100 non-null    float64
 3   superstore.no.transactions  100 non-null    int64  
 4   superstore.total.spend      100 non-null    float64
 5   gender                      100 non-null    object 
 6   income                      100 non-null    float64
 7   affluency                   100 non-null    object 
 8   Region                      100 non-null    object 
dtypes: float64(3), int64(3), object(3)
memory usage: 7.2+ KB


In [6]:
# Summary statistics using df.describe()

df.describe().astype(int)    

customer.id  express.no.transactions  express.total.spend  \
count          100                      100                  100   
mean         48808                       50                  834   
std          28244                       17                  898   
min           1668                        0                    0   
25%          28284                       40                  260   
50%          42780                       52                  620   
75%          74870                       62                 1092   
max          98018                       87                 6217   

       superstore.no.transactions  superstore.total.spend  income  
count                         100                     100     100  
mean                           75                   38924   25035  
std                            16                    9388    7775  
min                            22                   30002    5793  
25%                            65                   32265   20226  
50%                            74                   36049   25877  
75%                            86                   42026   30572  
max                           118                   73988   42847

In [7]:
# Summary statistcs for object data

df.describe(include=['object'])

gender affluency      Region
count      100       100         100
unique       2         5          11
top     Female       Mid  South East
freq        58        57          20

- The information above tells us "gender" has two unique values/classification. One of which is female and female appears 58 times in the data set. 
- Same for the othe column, affluency and Region
- affluency has 5 unique classifications and Region has 11 unique classifications in the data set, one of which is South East

In [8]:
print("Gender:",df.gender.unique(),"\n")
print("Affluency:",df.affluency.unique(),"\n")
print("Regions:",df.Region.unique())

Gender: ['Female' 'Male'] 

Affluency: ['Mid' 'High' 'Low' 'Very High' 'Very Low'] 

Regions: ['West Midlands' 'North West' 'Yorkshire & Humber' 'London' 'East'
 'South East' 'South West' 'North East' 'Central' 'East Midlands'
 'Central England']


In [9]:
#Create a dummy Variable Female from Gender

dummy = pd.get_dummies(df["gender"])
dummy.head()

Female  Male
0       1     0
1       0     1
2       0     1
3       0     1
4       0     1

In [10]:
df = df.merge(dummy[["Female"]], left_index=True, right_index=True)
df.head()

customer.id  express.no.transactions  express.total.spend  \
0         1668                       54              1559.20   
1         1827                       59               619.05   
2         3003                       59               211.50   
3         4398                       58              1118.04   
4         5055                       47              1318.61   

   superstore.no.transactions  superstore.total.spend  gender        income  \
0                          60                30874.62  Female  18901.724734   
1                          78                36466.45    Male  24464.981612   
2                          91                45903.40    Male  31263.210081   
3                          70                32175.17    Male  19742.126261   
4                          62                31109.30    Male  16165.819721   

  affluency              Region  Female  
0       Mid       West Midlands       1  
1      High          North West       0  
2      High          North West       0  
3       Low  Yorkshire & Humber       0  
4       Mid              London       0

In [11]:
#Create a dummy variable London from Region

dummy2 = pd.get_dummies(df["Region"])
dummy2.head()

Central  Central England  East  East Midlands  London  North East  \
0        0                0     0              0       0           0   
1        0                0     0              0       0           0   
2        0                0     0              0       0           0   
3        0                0     0              0       0           0   
4        0                0     0              0       1           0   

   North West  South East  South West  West Midlands  Yorkshire & Humber  
0           0           0           0              1                   0  
1           1           0           0              0                   0  
2           1           0           0              0                   0  
3           0           0           0              0                   1  
4           0           0           0              0                   0

In [12]:
df = df.merge(dummy2[["London"]], left_index=True, right_index=True)
df

customer.id  express.no.transactions  express.total.spend  \
0          1668                       54              1559.20   
1          1827                       59               619.05   
2          3003                       59               211.50   
3          4398                       58              1118.04   
4          5055                       47              1318.61   
..          ...                      ...                  ...   
95        93783                       39                48.04   
96        94420                       57              1144.88   
97        95503                       46                63.14   
98        97005                       23                69.48   
99        98018                       52              2494.40   

    superstore.no.transactions  superstore.total.spend  gender        income  \
0                           60                30874.62  Female  18901.724734   
1                           78                36466.45    Male  24464.981612   
2                           91                45903.40    Male  31263.210081   
3                           70                32175.17    Male  19742.126261   
4                           62                31109.30    Male  16165.819721   
..                         ...                     ...     ...           ...   
95                          92                53710.83  Female  35899.146160   
96                          72                32005.10  Female  23796.034048   
97                          89                52975.30    Male  38349.705738   
98                          99                52350.80  Female  33778.457469   
99                          49                30209.20  Female  12551.396826   

   affluency              Region  Female  London  
0        Mid       West Midlands       1       0  
1       High          North West       0       0  
2       High          North West       0       0  
3        Low  Yorkshire & Humber       0       0  
4        Mid              London       0       1  
..       ...                 ...     ...     ...  
95       Low          North West       1       0  
96       Mid              London       1       1  
97       Low              London       0       1  
98       Mid       East Midlands       1       0  
99       Low              London       1       1  

[100 rows x 11 columns]

In [13]:
# Summary of statistics of all variables

df.describe().astype(int)

customer.id  express.no.transactions  express.total.spend  \
count          100                      100                  100   
mean         48808                       50                  834   
std          28244                       17                  898   
min           1668                        0                    0   
25%          28284                       40                  260   
50%          42780                       52                  620   
75%          74870                       62                 1092   
max          98018                       87                 6217   

       superstore.no.transactions  superstore.total.spend  income  Female  \
count                         100                     100     100     100   
mean                           75                   38924   25035       0   
std                            16                    9388    7775       0   
min                            22                   30002    5793       0   
25%                            65                   32265   20226       0   
50%                            74                   36049   25877       1   
75%                            86                   42026   30572       1   
max                           118                   73988   42847       1   

       London  
count     100  
mean        0  
std         0  
min         0  
25%         0  
50%         0  
75%         0  
max         1

### (a) Report all quartiles of the variable ‘income’ and discuss their meaning.

- let Q represent quartiles
- Q1 = 20,226. Q1 is the same as 25 percentile(%)
- Q2 = 25,877. Q2 is also known as the median or 50 percentiles(%)
- Q3 = 30,572. Q3 also same as 75%, percentile



Q1 tells us that 25% of customer income are less than £20,226 and 75% had income greater.  Q2 is the 50th percentile and shows that 50% of the incomes are less than £25,877 and 50% of income are above £25,877. Q3 reveals that 75% of income are less than £30,572 and 25% are greater than £30,572

### (b) Is there a different spending pattern in superstores compared to express stores?

Yes.

Customers appear to spend more money in superstore than the express store. Superstore had on average, a total spend of £38,924 plus or minus £9,388 from each customer, while the express store had an average total spend of £834, plus or minus £898

The different spending pattern might be due to the superstore having a higher value of items like gadgest etc on their shelves. So it makes sense.

### c) What % of the sample consists of females and what % of the sample consists of consumers from London?

In [16]:
# Percentage of females

f = df["Female"].sum()

Count = df["gender"].count()

print("The sample consists of {} % of females.".format((f/Count)*100))

The sample consists of 57.99999999999999 % of females.


In [15]:
#Consumers from London
Lon = df["London"].sum()

print("\n {} % of the sample are from London".format((Lon/Count)*100))


 17.0 % of the sample are from London


### Calculate and report the average total spending between females and males in superstores, and the average total spending between Londoners and non-Londoners in express stores. Discuss your findings

In [20]:
# Average total spend between male and females

df[["superstore.total.spend", "gender", "income"]].groupby("gender").mean().astype(int)

superstore.total.spend  income
gender                                
Female                   39640   25964
Male                     37937   23752

- Females have an average total spending of £39,640
- Males have an average of £37,937

The total spending of females on average at superstores is slightly above that of the males, which makes sense since the females that spent at the superstores had slightly more income on average, than their male counterparts

In [21]:
# Average total spend for Londoners

df[["Region","express.total.spend","income"]].groupby("Region").mean().astype(int)

express.total.spend  income
Region                                         
Central                             623   24609
Central England                    1376   28669
East                                655   26622
East Midlands                       486   27217
London                             1244   21296
North East                         1014   21721
North West                          514   26750
South East                          826   25041
South West                         1228   26153
West Midlands                       796   25520
Yorkshire & Humber                  575   26434

- Londoners have an average total spending of £1,244. This comes third in line to Central England and South West who have an average total spend of £1,376 and £1,228 respectively.

Not much can be inferred from this except that Londoners were the 3rd highest spenders on average at the express superstores. Other regions save for Central England and South West, seemed to have low total spending in express stores on average.